src/main/java/by/tms/eshop/service/impl/CartServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package by.tms.eshop.service.impl;
+
+import by.tms.eshop.dto.LocationDto;
+import by.tms.eshop.dto.ProductDto;
+import by.tms.eshop.repository.CartRepository;
+import by.tms.eshop.service.CartService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+
+    private final CartRepository cartRepository;
+
+    @Transactional
```

Comment:
если убрать анатацию? то не будет в транзакции?

Reply:
Если убрать аннотацию, то не будет синхронизации между транзакцией спринга и хайбернэйта и получаем  "Resolved [org.springframework.dao.InvalidDataAccessApiUsageException: No EntityManager with actual transaction available for current thread - cannot reliably process 'persist' call]". Т.е. entityManager не знает о транзакции спринга, так как хайбернейт не управляет транзакцией автоматически: https://habr.com/ru/articles/682362/ =)

Reply:
я просто думал что у вас jpa метод и один только запрос в базу. если спринг дата стартер подключаете то фактически спирнг бут настроит все за вас. и любые запросы будут идти в транзакции. если вам надо будет несколько запросов отаправить в ттранзакции тогда просто помечаем анатацией @Transactional и @EnableTransactionManagement даже не нужна будет.

```
Probably you're also using Spring Data. Calls on Spring Data repositories are by default surrounded by a transaction, even without @EnableTransactionManagement. If Spring Data finds an existing transaction, the existing transaction will be re-used, otherwise a new transaction is created.

@Transactional annotations within your own code, however, are only evaluated when you have @EnableTransactionManagement activated (or configured transaction handling some other way).

You can easily trace transaction behavior by adding the following property to your application.properties:

logging.level.org.springframework.transaction.interceptor=TRACE
```

Reply:
https://stackoverflow.com/questions/40724100/enabletransactionmanagement-in-spring-boot

Reply:
https://stackoverflow.com/questions/40046116/spring-boot-with-spring-transactional-works-without-enabling-transactional-mana